packages/itwinui-react/src/styles.js/styles.module.css

```diff
@@ -8,12 +8,6 @@
 @import '@itwin/itwinui-variables' layer(itwinui.v3);
 @import '@itwin/itwinui-css';
 
-@layer itwinui.v3 {
-  .iui-root {
-    --_iui-v3-loaded: yes;
```

Comment:
Made an important update in [`45a249e`](https://github.com/iTwin/iTwinUI/pull/2106/commits/45a249e70f2f1946383a4e4030107236328078b0) to dynamically generate this property. It's important because the CSS between two 3.x versions is fully incompatible, so they all need to be loaded separately.

This actually highlights a larger issue that was present prior to this PR: an application could load two different v3 minors of iTwinUI. Best case, they end up with two copies of conflicting CSS. Worst case, they end up with missing CSS for classes that exist in the DOM but don't exist in the stylesheet that was loaded (e.g. `iui-field` was a new class added in 3.10.0, but the user might have only loaded 3.9.0). Of course it's not a problem if users ensure there is only copy per major version of iTwinUI, but after this PR it's fully solved on our end.

Reply:
Coincidentally, a user reported today that the recent change to use `<button>` in `MenuItem` (#2021) broke their visuals because of a CSS version mismatch.

![](https://github.com/iTwin/iTwinUI/assets/9084735/79540de7-89aa-4089-b4a8-1c212e62e2c4)

This PR should prevent such situations in the future.